Automattic/wp-calypso
# Update/editor welcome tour tracking and fixes

### Changes proposed in this Pull Request

![Screen Shot 2020-12-16 at 16 55 40](https://user-images.githubusercontent.com/5665959/102310683-98375a80-3fbf-11eb-810f-1364a4168aa3.png)


Add tracking to [Welcome Tour ](https://github.com/Automattic/wp-calypso/issues/47690)

- [x] When the tour modal opens
- [x] When a user dismisses the tour modal
- [x] When user minimizes the tour
- [x] For each slide, the slide number, whether it is the last slide and a unique string, such as the slide header
- [x] Tracking events for the thumbs up/down rating

### Testing instructions
* Sync Editing Toolkit to your sandbox (e.g. pull down this PR, then [sync to sandbox](PCYsg-ly5-p2#wpcom-sandbox)
* Add `define( 'SHOW_WELCOME_TOUR', true );`to your ` /home/wpcom/public_html/wp-content/mu-plugins/0-sandbox.php` file. This will force the Welcome Tour to show every time the editor loads.

* View the Welcome Tour model opening in the Editor 
* In DevTools Network request tab, verify tracking network request (see screenshot) for each of the tracking events.  And 
  * `calypso_editor_wpcom_tour_open`   
      - data: `{is_gutenboarding:}`
  * `calypso_editor_wpcom_tour_dismiss` 
      - data: `{is_gutenboarding: , slide_number: , action: }`
  * `calypso_editor_wpcom_tour_slide_view` (fired on each slide, but NOT after a maximize event)
     - data:  `{slide_number:, is_last_slide:, slide_heading:, is_gutenboarding:}`
  * `calypso_editor_wpcom_tour_minimize`
     - data `{slide_number:,  is_gutenboarding:}`
  * `calypso_editor_wpcom_tour_minimize`
     - data `{slide_number:,  is_gutenboarding:}`
  * `calypso_editor_wpcom_tour_rate`
       - data `{thumps_up: , is_gutenboarding:}`


Ensure the following during testing
* slide view after maximizing _should not_ be tracked
* slide view going backward _should be_ tracked
